src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1539,18 +1549,25 @@ protected long getNextDelayInMillis() {
     protected long getDelayEstimateInMs() {
       return nextRunTime - Clock.lastKnownForwardProgressingMillis();
     }
-    
+
     @Override
-    public void executing() {
+    public boolean canExecute() {
       if (canceled) {
-        return;
+        return false;
+      }
+      synchronized (queueSet.scheduleQueue.getModificationLock()) {
```

Comment:
The remove is atomic.  The reason we have to lock here is because if we have a recurring task, we need to make sure it is both removed, and re-added to the queue atomically.  Otherwise we could get into a condition where we may not be able to remove the recurring task because it can not be found in the queue.

That said, I will see if a reposition call is more efficent.